simtools/model/mirrors.py

```diff
@@ -136,23 +131,37 @@ def get_single_mirror_parameters(self, number):
 
         Returns
         -------
-        (pos_x, pos_y, diameter, flen, shape): tuple of float
-            X, Y positions, diameter, focal length and shape.
+        (pos_x, pos_y, mirror_diameter, focal_length, shape_type): tuple of float
+            X, Y positions, mirror_diameter, focal length and shape_type.
         """
 
-        if number > self.number_of_mirrors - 1:
-            self._logger.error("Mirror number is out range")
+        def get_mirror_table_mask():
+            if isinstance(self.mirror_table["mirror_id"][0], np.str_):
+                mask = self.mirror_table["mirror_id"] == "id=" + str(number)
```

Comment:
Following the long discussion we had today about strings, you should use f-string here `f"id={number}"`

Reply:
Changing.